orientdb/src/main/scala/akka/stream/alpakka/orientdb/OrientDBSinkSettings.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.orientdb
+
+import com.orientechnologies.orient.core.db.OPartitionedDatabasePool
+
+//#sink-settings
+final case class OrientDBSinkSettings(oDatabasePool: OPartitionedDatabasePool,
+                                      maxPartitionSize: Int = Runtime.getRuntime.availableProcessors(),
+                                      maxPoolSize: Int = -1,
+                                      maxRetry: Int = 1,
+                                      retryInterval: Int = 5000,
+                                      bufferSize: Int = 10)
+//#sink-settings
+{
+
+  def this(oDatabasePool: OPartitionedDatabasePool) =
+    this(oDatabasePool, Runtime.getRuntime.availableProcessors(), -1, -1, 5000, 10)
+
+  def withMaxPartitionSize(maxPartitionSize: Int): OrientDBSinkSettings =
+    copy(
```

Comment:
You can simply do `copy(maxPartitionSize = maxPartitionSize)` all the other fields are automatically copied (as the method name says).

The same applies to all the other `withXXX` methods